lma-ai-stack/source/lambda_functions/share_meeting/index.py

```diff
@@ -0,0 +1,167 @@
+                }
+            ]
+        else:
+            response = ddbTable.query(
+                KeyConditionExpression=Key('PK').eq(pk)
+            )
+            items = response['Items']
+            
+            while 'LastEvaluatedKey' in response:
+                response = ddbTable.query(
+                    KeyConditionExpression=Key('PK').eq(pk),
+                    ExclusiveStartKey=response['LastEvaluatedKey']
+                )
+                items.extend(response['Items'])
+
+        updated_count = 0
+        for item in items:
+            ddbTable.update_item(
+                Key={
+                    'PK': item['PK'],
+                    'SK': item['SK']
+                },
+                UpdateExpression="SET SharedWith = :val",
+                ExpressionAttributeValues={':val': recipients}
+            )
+            updated_count += 1
+
+    except ClientError as err:
+        logger.error("Error updating recipients for %s: %s",
+                     err.response['Error']['Code'], err.response['Error']['Message'])
+        raise
+    else:
+        return updated_count
+
+def get_call_metadata(callid):
+    pk = 'c#'+callid
+    print(f"Call metadata PK: {pk}")
+    try:
+        metadata = ddbTable.get_item(
+            Key={'PK': pk, 'SK': pk},
+            TableName=LCA_CALL_EVENTS_TABLE
+            )
+    except ClientError as err:
+        logger.error("Error getting metadata from LCA Call Events table %s: %s",
+                     err.response['Error']['Code'], err.response['Error']['Message'])
+        raise
+    else:
+        return metadata['Item']
+
+def get_transcripts(callid):
+    pk = 'trs#'+callid
+    print(f"Call Transcript PK: {pk}")
+
+    try:
+        response = ddbTable.query(KeyConditionExpression=Key('PK').eq(pk), FilterExpression=(
+            Attr('Channel').eq('AGENT') | Attr('Channel').eq('CALLER')) & Attr('IsPartial').eq(False))
+        # response = ddbTable.query(KeyConditionExpression=Key('PK').eq(pk))
+    except ClientError as err:
+        logger.error("Error getting transcripts from LCA Call Events table %s: %s",
+                     err.response['Error']['Code'], err.response['Error']['Message'])
+        raise
+    else:
+        # print(response['Items'])
+        return response['Items']
+
+def lambda_handler(event, context):
+    print("Received event: " + json.dumps(event, indent=2))
+    request_owner = get_owner_from_jwt(event['accessToken'], True)
+    print("DECODED JWT", request_owner)
+
+
+    data = json.loads(json.dumps(event))
+    calls = data['calls']
+    
+    for call in calls:        
+        callid = call['callId']
+        shardPK = call['PK']
+        shardSK = call['SK']
+
+        print("CallID: ", callid, shardPK, shardSK)
+        transcripts = get_transcripts(callid)
+        metadata = get_call_metadata(callid)
+        print("Fetch Transcript response:", metadata)
+        if(metadata.get('Owner', '') != request_owner):
```

Comment:
I was wondering if we could/should do this check earlier, in the UI to provide immediate feedback to user that they cannot update a call they don't own.

Reply:
Agree that we can save a call to the Lambda function by doing this. Maybe address this during refactor next week with other changes?